### PR TITLE
Refactor the code to remove duplications

### PR DESCRIPTION
**Pull request by marceloamaral**
This pull request addresses the issues raised in both #630 and #663, which describe the problem of significant code duplication leading to persistent lint complaints. Code duplication not only triggers lint warnings but also introduces challenges in maintenance and raises the risk of errors.

The key changes in this PR include:
- [x] Removing code duplication related to the metric and power packages related to processes, containers, and virtual machine
- [x] Reorganizing the metric collection packages by restructuring them into folders
- [x] Enhancing the collection and estimation of power metrics on a per-process basis, followed by aggregating these metrics per container or virtual machine
- [x] Fixing the VM ID collector to ensure consistent resolution of both container and VM IDs
- [x] Refactoring the metric map for better consistency, consolidating all metrics into energy or resource utilization maps, instead of having multiple maps
- [x] Introducing a test utility to ensure uniform creation of mock process and node metrics across all tests
- [x] Fixing BCC code to collect cgroup ID if the OS is configured with cgroup v1
- [x] Removing many unused functions
- [x] Adding the option to disable container and VM metrics.
- [x] Transferring the Prometheus code to the metrics package, aligning with the convention used in other Golang projects. Consequently, the previous container and process metrics are now part of the stats package. Additionally, a class has been created to generate different Prometheus collector (exporter) for process, container, VM, and node metrics, thereby reducing extensive code duplication
- [x] Renaming the power package to "sensors" since it is not collecting only power metrics anymore
- [x] Addressing and fixing test issues

### CODE REVIEW

**Comment by aharivel** [comment]
Hi @marceloamaral ,
Thanks for this huge work!
I'm going to test the VM metrics collection part and ensure everything is working as expected as soon as the checks are successful. 

Anthony

File: pkg/collector/energy/node_energy_collector.go [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+}
+
+// UpdateNodeEnergyMetrics updates the node energy consumption of each component
+func UpdateNodeEnergyMetrics(nodeStats *stats.NodeStats) {
+	var wgNode sync.WaitGroup
+	wgNode.Add(2)
+	go UpdateNodeComponentsEnergy(nodeStats, &wgNode)
+	go UpdateNodeGPUEnergy(nodeStats, &wgNode)
+	wgNode.Wait()
+	// update platform power later to avoid race condition when using estimation power model
```
Comment (sthaha) [inline pkg/collector/energy/node_energy_collector.go:102]: nice! thanks for the explanation

**Comment by rootfs** [comment]
there is an e2e test failure 
```
• [FAILED] [0.002 seconds]
Metrics check should pass Check node level metrics for details [It] Checking kepler_node_core_joules_total
/home/runner/work/kepler/kepler/e2e/integration-test/e2e_metric_test.go:143

  [FAILED] Expected
      <bool>: false
  to be true
  In [It] at: /home/runner/work/kepler/kepler/e2e/integration-test/e2e_metric_test.go:125 @ 11/28/23 10:23:35.35
```

**Comment by rootfs** [comment]
the benchmark got an error:
```
*** Test killed with quit: ran too long (11m0s).
exit status 2
FAIL	github.com/sustainable-computing-io/kepler/pkg/model	660.112s
```

**Comment by marceloamaral** [comment]
I needed to add back the libpf installation as described in the issue #1128

**Comment by marceloamaral** [comment]
@rootfs and @aharivel it is ready to review again.

**Review by rootfs — APPROVED** [review]

**Comment by rootfs** [comment]
@marceloamaral let's make a 0.7.2 release first and merge this PR to 0.7.3, wdyt?

cc @husky-parul

File: pkg/model/model.go [hunk 2]
```diff
@@ -56,29 +56,29 @@ type PowerMoldelInterface interface {
...
-func CreatePowerEstimatorModels(containerFeatureNames, systemMetaDataFeatureNames, systemMetaDataFeatureValues []string) {
+func CreatePowerEstimatorModels(processFeatureNames, systemMetaDataFeatureNames, systemMetaDataFeatureValues []string) {
 	config.InitModelConfigMap()
-	CreateContainerPowerEstimatorModel(containerFeatureNames, systemMetaDataFeatureNames, systemMetaDataFeatureValues)
-	CreateProcessPowerEstimatorModel(containerFeatureNames, systemMetaDataFeatureNames, systemMetaDataFeatureValues)
-	// Node power estimator uses the container features to estimate node power, expect for the Ratio power model that contains additional metrics.
-	CreateNodePlatformPoweEstimatorModel(containerFeatureNames, systemMetaDataFeatureNames, systemMetaDataFeatureValues)
-	CreateNodeComponentPoweEstimatorModel(containerFeatureNames, systemMetaDataFeatureNames, systemMetaDataFeatureValues)
+	CreateProcessPowerEstimatorModel(processFeatureNames, systemMetaDataFeatureNames, systemMetaDataFeatureValues)
+	CreateProcessPowerEstimatorModel(processFeatureNames, systemMetaDataFeatureNames, systemMetaDataFeatureValues)
```
Comment (vimalk78) [inline pkg/model/model.go:78]: `CreateProcessPowerEstimatorModel` is this function call duplicated?

Comment (marceloamaral) [reply pkg/model/model.go:78]: Thanks @vimalk78 for finding this.
I created the PR #1234 to fix it.

**Pull request merged**